packages/cli/src/build/request.rs

```diff
@@ -552,24 +552,36 @@ impl BuildRequest {
             .iter()
             .any(|dep| dep.name == "dioxus");
 
+        // Infer the renderer from platform argument if the platform argument is "native" or "desktop"
+        let renderer = args.renderer.or(match args.platform {
+            Some(PlatformArg::Desktop) => Some(ClientRenderer::Webview),
+            Some(PlatformArg::Native) => Some(ClientRenderer::Native),
+            _ => None,
+        });
+
         let mut features = args.features.clone();
         let mut no_default_features = args.no_default_features;
 
         let platform: Platform = match args.platform {
-            Some(platform) => match enabled_platforms.len() {
-                0 => platform,
+            Some(platform_arg) => match enabled_platforms.len() {
+                0 => Platform::from(platform_arg),
 
                 // The user passed --platform XYZ but already has `default = ["ABC"]` in their Cargo.toml or dioxus = { features = ["abc"] }
                 // We want to strip out the default platform and use the one they passed, setting no-default-features
                 _ => {
                     features.extend(Self::platformless_features(main_package));
                     no_default_features = true;
-                    platform
+                    Platform::from(platform_arg)
                 }
             },
-            None if !using_dioxus_explicitly => Platform::autodetect_from_cargo_feature("desktop").unwrap(),
+            None if !using_dioxus_explicitly => Platform::TARGET_PLATFORM.unwrap(),
             None => match enabled_platforms.len() {
-                0 => return Err(anyhow::anyhow!("No platform specified and no platform marked as default in Cargo.toml. Try specifying a platform with `--platform`").into()),
+                0 => match renderer {
+                    Some(_) => Platform::TARGET_PLATFORM.unwrap(),
+                    None => Platform::TARGET_PLATFORM.unwrap(),
+                    // TODO: should we always have a default
+                    // None => return Err(anyhow::anyhow!("No platform specified and no platform marked as default in Cargo.toml. Try specifying a platform with `--platform`").into()),
```

Comment:
@jkelleyrtp This was the only thing I wasn't sure about. Should we pick a default platform if none is specified at all? Either way I ought to either remove or reinstate the commented out code.